app/api/routers/breeze_buddy/users/handlers.py

```
@@ -0,0 +1,222 @@
+            merchant_ids=merchant_data.merchant_ids,
+            shop_identifiers=merchant_data.shop_identifiers,
+            is_active=merchant_data.is_active,
+        )
+    except asyncpg.UniqueViolationError as e:
+        # Handle race condition where concurrent request passed validation checks
+        error_msg = str(e)
+        if "username" in error_msg:
+            raise HTTPException(
+                status_code=status.HTTP_409_CONFLICT,
+                detail="Username already exists",
+            )
```

Ruff (0.14.14)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)